From 86e9235918a715095a1f5bb1c1db28fae7fca22b Mon Sep 17 00:00:00 2001
From: Ulrich Drepper <drepper@gmail.com>
Date: Wed, 19 Jan 2011 15:50:05 -0500
Subject: [PATCH] Fix decoding of canonical name in getaddrinfo.

---
 ChangeLog                   |    5 +++++
 sysdeps/posix/getaddrinfo.c |    1 +
 2 files changed, 6 insertions(+), 0 deletions(-)

2010-10-01  Andreas Schwab  <schwab@redhat.com>

	* sysdeps/posix/getaddrinfo.c (gaih_inet): Don't discard result of
	decoding ACE if AI_CANONIDN.

diff --git a/sysdeps/posix/getaddrinfo.c b/sysdeps/posix/getaddrinfo.c
index 126a09e..c61c72a 100644
--- a/sysdeps/posix/getaddrinfo.c
+++ b/sysdeps/posix/getaddrinfo.c
@@ -965,6 +965,7 @@ gaih_inet (const char *name, const struct gaih_service *service,
 		   make a copy.  */
 		if (out == canon)
 		  goto make_copy;
+		canon = out;
 	      }
 	    else
 #endif
-- 
1.7.5.rc3

